### PR TITLE
Adds uninstall task for removing plugins based on the plugin name/key

### PR DESCRIPTION
**Pull request by andrenpaes**
* Deletes plugin from the fs, asking for confirmation if multiple version are installed
* Cleans up sha1.yml and plugin_identifiers.json files

I'm still working on the specs.

### CODE REVIEW

File: kpm/lib/kpm/uninstaller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+      plugins_installation_path = File.join(destination, 'plugins')
+      @plugins_manager = PluginsManager.new(plugins_installation_path, @logger)
+
+      sha1_file_path = File.join(destination, KPM::BaseInstaller::SHA1_FILENAME)
+      @sha1checker = KPM::Sha1Checker.from_file(sha1_file_path, @logger)
+    end
+
+    def uninstall_plugin(plugin, force = false)
+      plugin_info = find_plugin(plugin)
```
Comment (sbrossie) [inline kpm/lib/kpm/uninstaller.rb]: Shouldn't the second arg be the version (`options[:version]`) ?

**Comment by sbrossie** [comment]
Looks good. Unclear whether there is still more check-in required for tests and also status about deleting a specific plugin version

File: kpm/lib/kpm/uninstaller.rb (hunk 1 above)
Comment (andrenpaes) [inline kpm/lib/kpm/uninstaller.rb]: Actually, the tasks.rb has an issue, it should be:
```ruby
say 'Done!' if Uninstaller.new(options[:destination]).uninstall_plugin(plugin, options[:force])
```

**Comment by andrenpaes** [comment]
My first implementation had a `version` option, but the removal of a specific version is a little bit trickier, so simplified it a bit. Let me try to summarize the issue:

If multiple versions of a plugin are installed, only the latest version is present in plugin_identifiers.json file. Once we run `uninstall` for the latest version, the file system is cleaned, and the entries in `plugin_identifiers.json` and `sha1.yml` files are removed.

At this point, there are no more entries for this plugin in `plugin_identifiers.json`, but there still an entry in `sha1.yml`. So we ended up leaving the system in a inconsistent state. If we try to uninstall another version of the plugin, we'd be able to remove the files from the file system, but the info to find the entry in the sha1.yml file is gone (group id, artifact id, etc, which would be in `plugin_identifiers.json`).

So in order to make sure uninstall leaves things in a, as much as possible, consistent state, it removes all the versions and their respective entries in the descriptors.

I'm ok with switching back to having the version as a command option, it's up to you guys.

Finally, I think better solution for this problem would be keeping all the versions installed in `plugin_identifiers.json`:
```json
{
  "adyen": [{
      "plugin_name": "adyen-plugin",
      "language": "java",
      "group_id": "org.kill-bill.billing.plugin.java",
      "artifact_id": "adyen-plugin",
      "packaging": "jar",
      "classifier": null,
      "version": "0.2.1"
    },
    {
      "plugin_name": "adyen-plugin",
      "language": "java",
      "group_id": "org.kill-bill.billing.plugin.java",
      "artifact_id": "adyen-plugin",
      "packaging": "jar",
      "classifier": null,
      "version": "0.2.0"
    }
  ],
  ...
}
```
Or something like that. But it's definitely a bigger change

**Comment by sbrossie** [comment]
@andrenpaes Don't worry about the version option if that gets too complicated. It's a nice to have, not super critical.

**Pull request merged**